### PR TITLE
Cisco NSO module migration

### PR DESCRIPTION
**Pull request by mamullen13316**
##### SUMMARY
Migrate the Cisco NSO modules to the cisco.nso namespace

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
- nso_config
- nso_verify
- nso_query
- nso_action
- nso_show

##### ADDITIONAL INFORMATION
The new collection is active on Galaxy here: https://galaxy.ansible.com/cisco/nso and on Github: https://github.com/CiscoDevNet/ansible-nso. This PR is to migrate the modules in community.network  over to the new collection.  I tried to follow along with the previous Onyx PR (#74 ) which I believe was a similar migration, but please let me know if there is anything that I am missing or needs to be corrected.

### CODE REVIEW

**Review by felixfontein — CHANGES_REQUESTED** [review]
Adding negative review to avoid accidental merge.

**Comment by mamullen13316** [comment]
Hello @felixfontein,  are there any changes required for this PR to be merged and when might it be merged?   Also, will merging this PR take care of re-pointing the documentation on docs.ansible.com to the new collection or is there something separate that needs to be done to accomplish that?

**Comment by felixfontein** [comment]
@mamullen13316 the PR is probably merged for 2.0.0, but with some modifications, especially w.r.t the deprecations of the redirects. We're still discussing how to do post-2.10.0 migrations in the community meeting. We've accumulated a bunch of these right now, and I think we're close to finally know how to make the move.

**Comment by felixfontein** [comment]
For an example of how this PR should look like, see both f896c2986c024436a9e12ea6f9e5bbc828907dc5 (main removal) and dbae7da6bcfeea9c1c64060b577dec72b8f9f2a0 (fixes changelog fragment, removes changelog fragments).

Please note that this can only be merged once the new collection is included in Ansible 2.10.

**Comment by mamullen13316** [comment]
> For an example of how this PR should look like, see both f896c2986c024436a9e12ea6f9e5bbc828907dc5 (main removal) and dbae7da6bcfeea9c1c64060b577dec72b8f9f2a0 (fixes changelog fragment, removes changelog fragments).
> 
> Please note that this can only be merged once the new collection is included in Ansible 2.10.

Thanks I will review those commits.   What needs to be done to have the new collection included in Ansible 2.10?

**Comment by felixfontein** [comment]
The conditions on the checklist https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst need to be satisfied, and it needs to be added to the Ansible build system (https://github.com/ansible-community/ansible-build-data/blob/main/2.10/ansible.in). The main ticket for discussing this should be https://github.com/CiscoDevNet/ansible-nso/issues/4

**Comment by mamullen13316** [comment]
Latest commit a2e7066 contains changes to align with f896c2986c024436a9e12ea6f9e5bbc828907dc5 and dbae7da6bcfeea9c1c64060b577dec72b8f9f2a0 as mentioned in the above comments.  I also merged the changes with a `git pull` to my local repo in order to incorporate the changes over the last 28 days, and resolved a merge conflict I had with meta/runtime.yml.  Please let me know if this looks ok.

**Comment by felixfontein** [comment]
There are multiple nso-related files that are still around:
1. ./plugins/module_utils/network/nso/nso.py (and the directory containing it)
2. ./plugins/modules/network/nso/nso_action.py
3. ./plugins/doc_fragments/nso.py

Also there's no redirect for the doc_fragments plugin (for examples, see https://github.com/ansible-collections/community.general/blob/main/meta/runtime.yml#L503).

**Comment by mamullen13316** [comment]
Not sure how those files came back, they were deleted in the initial commit for this PR (7a0fc45).  It looks like maybe during the merge.  I checked just to make sure there hadn't been any changes to them since I initially opened the PR,  and there haven't been any changes for 3 months.  So scratching my head as to why they came back.  In any case,  this should take care of them as well as adding the redirect for the doc_fragment in meta/runtime.yml.

File: meta/runtime.yml [hunk 1]
```diff
@@ -156,3 +168,7 @@ plugin_routing:
   terminal:
     routeros:
       redirect: community.routeros.routeros
+doc_fragments:
+  nso:
+    redirect: cisco.nso.nso
```
Comment (felixfontein) [inline meta/runtime.yml:173]: This is one level too far up:
```suggestion
  doc_fragments:
    nso:
      redirect: cisco.nso.nso
```

**Review by felixfontein — CHANGES_REQUESTED** [review]
Looks good!

Adding a negative review to prevent premature merge. Once the collection has been added to ansible.in, this can be merged. See [here](https://github.com/ansible-collections/overview/issues/128#issuecomment-730620776) and [here](https://github.com/CiscoDevNet/ansible-nso/issues/4) for more details.

**Review by felixfontein — APPROVED** [review]
https://github.com/ansible-community/ansible-build-data/pull/41 has been merged

**Pull request merged**